### DISCUSSION TITLE
Welcome to the CBS Researcher Toolkit!

### DISCUSSION THREAD

**Discussion by conorwild** [TheOwenLab/cbs-toolkit#1] [Announcements]
## 👋 Welcome!
  This is a place where we can have discussions about Cambridge Brain Sciences data! You can:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Ask for tech support.
  * Suggest new features.
  * Welcome others! Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by conorwild** [comment]
Anyways, my hope is that we can really make use of this platform to centralize information about working with CBS data for research projects. For example, I think the WIKI will eventually contain information about the tasks and their scores / features, best practices for analyzing CBS data, and how to approach these data.

Please take a look around! Explore, investigate, learn about all the tools at our disposal. Thanks!

**Comment by mike-battista** [comment]
Hey @conorwild , could I get added as a collaborator? I'm Mike, I help set up all the CBS Research studies, and my GitHub username is mike-battista . Thanks! :)

**Reply by conorwild** [reply]
Hey @mike-battista , invitation sent and you should be part of the CBS Dev team which will allow you to update the repository, WIKI etc.

**Comment by SajeevanS** [comment]
Hey Conor, this is Sajeevan (I work on AcT-Cog with Dr. Swartz) could I get added as a collaborator, so I can get read access for the CBS python image. My GItHub username is SajeevanS

**Reply by conorwild** [reply]
Hey @SajeevanS, invitation has been sent. I'll keep an eye on my notifications for the evening, so let me know how it goes or if you have any questions.

**Reply by SajeevanS** [reply]
Thank you Conor!

**Comment by jbrand9** [comment]
Hi Conor, I work with Dr. Norton and require access for the GH package (I think it's called). We will be using it for data analysis of traumatic brain injury and body patients. My username is jbrand9. Thanks!

**Comment by jbrand9** [comment]
Hi Conor, I'm having difficulty logging into to docker through the my terminal. Maybe you can help? Here is was I'm putting in.... set GH_PAT=(token provided on my account)
echo $GH_PAT | docker login ghcr.io -u jbrand9 --password-stdin This is the response I get: Error response from daemon: Get "https://ghcr.io/v2/": denied: denied

**Reply by conorwild** [reply]
Hey @jbrand9! First thing you should try is using `export GH_PAT=yourtokenvalue`, **not** `set GH_PAT=...`.  After that, if you type `echo $GH_PAT` on the terminal line, you should see it print out your token. If that works, try the next line `echo ... | docker login ...`

**Reply by jbrand9** [reply]
Hi Conor, so I tried using “export” initially but was met with an error that said something like “export command does not exist internally or externally”? That’s why I used “set” based on other forums and I got the response above. Also, do I put the export and echo commands within the same prompt or 2 lines. Sorry a bit confusing!

**Reply by conorwild** [reply]
What OS are you using, and what kind of terminal shell? Ok, here's a workaround - don't bother trying to set the environment variable first. Just pipe it directly into the login command like this:
`echo "ghp_tokenthisismadeup" | docker login ghcr.io -u jbrand9 --password-stdin`

Note, put the `"` around your token string.

**Reply by jbrand9** [reply]
Using Windows and command I believe? I’ll give that a try the next time I’m in the lab thank you!

**Reply by conorwild** [reply]
Ah that would explain it - I haven't tested on Windows much. The workaround should be OK though. I will be out of the office next week, so I will be slow getting back to you if you encounter any further issues in the short term. Good luck!

**Reply by jbrand9** [reply]
Hi Conor. Unfortunately, the work around you posted still does not work. The response I get after putting in the new token is: Error response from daemon: Get "https://ghcr.io/v2/": denied: denied.

**Reply by jbrand9** [reply]
Hi Conor, just wanted to follow up on these comments!

**Reply by conorwild** [reply]
Hey @jbrand9 - have you tried this?
`docker login ghcr.io -u jbrand9 --password ghp_123456789abcdefg`
Except, paste in your actual token value.

**Reply by jbrand9** [reply]
Hey Conor, entering in that command all on one line did not work. So instead, I did this....docker login ghcr.io -u jbrand9.... which I was then prompted for a password so I entered my token which was met with Login Succeeded. However, now when I put in....  pull ghcr.io/theowenlab/cbspython:latest ... I am given.... Error response from daemon: denied

**Reply by jbrand9** [reply]
Hey Conor, just checking in again. Still don't have access :(

**Reply by conorwild-test** [reply]
@jbrand9 , I have tested with multiple other accounts and collaborators - and it seems to be working. You are a member of the owenlab collaborators, and have access to the correct repository. Are you sure that you are logging into ghcr.io before you try the `docker pull..` command?  It will not work if you close that terminal window or start a new one. Also, it is helpful for debugging if you could copy and paste (or screenshot) your terminal history so that I can see the commands that you tried and the errors. Thanks.

**Reply by jbrand9** [reply]
Hi Conor, here are a couple screen shots.

![Attempt1](https://github.com/TheOwenLab/cbs-toolkit/assets/137221285/c31daa2d-391d-4ea8-9df3-fdbe21be6327)
![Attempt2](https://github.com/TheOwenLab/cbs-toolkit/assets/137221285/43a71c45-5682-4e94-aa8e-5fa52f20b800)
![Attempt3](https://github.com/TheOwenLab/cbs-toolkit/assets/137221285/5af2aedb-171c-448e-b052-7ad533101219)

**Reply by jbrand9** [reply]
Note in the third attempt, the username and password were inputted independently. I pasted the token where "password" is prompted, but it has been blanked out by the command terminal. However, after I pasted the token and pressed enter, this was the only time login was successful.

**Comment by conorwild** [comment]
Thanks for the screenshots - those help. In the first two, it looks like
the error is a result of the docker daemon (background service) is not
running. That’s ok, it sometimes takes a while to start up. You can check
if it’s ready by trying ‘docker run hello-world’.

The third login attempt works because docker is ready, but I’m not so sure
about why you are getting getting denied. Can you check your GitHub
Token, or make a new one, to make sure that have “read repositories”
selected? Also it should be a “classic” token, and not a fine grained one (or
whatever the other kind are called).

On Mon, Jul 31, 2023 at 7:36 AM jbrand9 ***@***.***> wrote:

> Note in the third attempt, I pasted the token but it has been blanked out
> by the command terminal. However, after I pasted the token and pressing
> enter, this was the only time Login was successful.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TheOwenLab/cbs-toolkit/discussions/1#discussioncomment-6595326>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMAX73AEKU627PN6TNEQFDXS6YHNANCNFSM57L2SS5A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Conor Wild
PhD, Research Associate
Creyos | The Owen Lab
Western University | London ON | N6A 5B7

www.owenlab.uwo.ca

www.creyos.com
www.creyos.com/science/creyos-research

**Reply by jbrand9** [reply]
Hey Conor, good news! I was able to make it work. Thanks for your help! I made the mistake of using a fine grained token. I switched to a classic one and all is well. I don't think it mentioned using a classic token on the CBS Python (Docker) page which might be helpful to others in the future! Thanks again :)

**Reply by jbrand9** [reply]
Hi again, Conor. I'm having issues getting the 6 files from the score calculator. I have ensured my login works and the image is successfully pulled. If you could look at the screenshot that would be helpful. 
![Script](https://github.com/TheOwenLab/cbs-toolkit/assets/137221285/c16cf18d-9ef8-4abc-843a-35ff19b6542c)

**Reply by conorwild** [reply]
Hi @jbrand9, just got back from a week away - it looks like you are making some progress. The issue in your most recent post is due to calling the program incorrectly. You are missing much of the docker "magic" at the start of the command. Try reading https://github.com/TheOwenLab/cbs-toolkit/wiki/CBS-Python-(Docker) , to see what the format of the commands should look like. Specifically, it should look something like this:
`docker run --rm -it -v $PWD:/tmp -w /tmp ghcr.io/theowenlab/cbspython:latest score_calculator ...`
Note, you might need to change `$PWD` because you are on windows. If that doesn't work, you can try a period `... -v .:/tmp -w /tmp ...` or `%cd%` like this: `... -v %cd%:/tmp -w /tmp ...`

**Reply by jbrand9** [reply]
Hi again Conor, I hope you had a good week. I don't understand what is being asked with directories and the terminal in step 3 of https://github.com/TheOwenLab/cbs-toolkit/wiki/cbs_score_calculator. I also did what you said and received the error below. FYI: the files are located on my desktop and are csv files etc. 
![CBSAttempt](https://github.com/TheOwenLab/cbs-toolkit/assets/137221285/6b38ca97-9b13-459c-84ac-a3235161ffbf)

**Reply by conorwild** [reply]
You should first navigate to your Desktop directory in the terminal, if that's where your files are located (currently, it looks like you are in `C:\Users\cogne`)

**Reply by jbrand9** [reply]
How do I go about navigating to the desktop with terminal? I have never done that before.

**Reply by conorwild** [reply]
Depending on your version of Windows, either `cd Desktop` or `cd OneDrive\Desktop`. You can check if you are in the correct directory by listing the contents of the folder, using the `dir` command. You should see your files.

**Reply by jbrand9** [reply]
SUCESS THANK YOU
